Bug fixes (the `simplify` branch)

DmitrySharabin opened the pull request:
### Summary

- Fix the C, C#, F#, Markdown, PHP, and VB.NET language definitions
- Fix the issue with deeply nested properties in `extend()`

DmitrySharabin committed:
[C] Fix transformation bug: `macro` should be inserted before `string`

DmitrySharabin committed:
[F#] Specify the base language

DmitrySharabin committed:
[extend] Deeply nested objects can be `undefined`

DmitrySharabin committed:
[C#] Specify the base language + remove unused imports

DmitrySharabin committed:
[PHP] Add missed tokens

DmitrySharabin commented on src/languages/c.ts:
```diff
@@ -37,48 +37,48 @@ export default {
-							{
-								// highlight the path of the include statement as a string
-								pattern: /^(#\s*include\s*)<[^>]+>/,
+					'macro': {
```
`macro` should be inserted before `string`, not that something should be inserted before `macro`.

DmitrySharabin commented on src/languages/aspnet.ts:
```diff
@@ -1,55 +1,44 @@
 import csharp from './csharp';
 import markup from './markup';
-import type { Grammar, GrammarOptions, LanguageProto } from '../types';
+import type { Grammar, GrammarToken, LanguageProto } from '../types';
```
By bad. I transformed this language incorrectly before.

DmitrySharabin committed:
[VB.NET] Fix language definition

DmitrySharabin committed:
[MD] Correctly capture the language code

DmitrySharabin commented on src/languages/markdown.ts:
````diff
@@ -95,7 +95,7 @@ export default {
 					// ```
 					pattern:
-						/^```(?<codeLanguage>[a-z-]+).+(?:\n|\r\n?)(?<codeBlock>[\s\S]*)?(?:\n|\r\n?)```$/i,
+						/^```(?<codeLanguage>[a-z-]+)(?:.+)?(?:\n|\r\n?)(?<codeBlock>[\s\S]*)?(?:\n|\r\n?)```$/i,
````
Cover cases when there is no space (only a line break) after the language code, and its last symbol might not be treated as part of the language code.

LeaVerou commented on src/languages/aspnet.ts:
```diff
@@ -1,55 +1,44 @@
-				},
-				$rest: 'csharp',
+	grammar ({ base }): Grammar {
+		const pageDirectiveInside: Grammar = {
```
Why did we rename this?

DmitrySharabin replied:
Hm, something is wrong with the changes I propose. I took the language definition from the `v2` branch, not the `simplify` branch. Let me remove it from the PR for now. We'll return to it later.

LeaVerou reviewed (approved):
See question but otherwise LGTM

DmitrySharabin commented:
> See question but otherwise LGTM

If you are referring to the [ASP.NET language and why we renamed the variable](https://github.com/PrismJS/prism/pull/3943#discussion_r2111947471), I removed this language from this PR to update it correctly later.
